features/steps/reusable_steps.rb

```diff
@@ -6,24 +6,20 @@
   Maze.config.captured_invalid_requests.delete(type.to_sym)
 end
 
-Then(/^on (iOS|macOS|watchOS), (.+)/) do |platform, step_text|
-  step(step_text) if platform.downcase == Maze::Helper.get_current_platform
+def platform_matches?(platform)
+  Maze::Helper.get_current_platform == platform.downcase
 end
 
-Then(/^on (\w+) ([0-9.]+) and later, (.+)/) do |test_platform, test_version, step_text|
-  actual_platform = Maze::Helper.get_current_platform
-  unless %w[ios macos].include? actual_platform
-    raise "Unexpected platform #{actual_platform}, this step can only handle ios and macos"
-  end
-
-  actual_version = Maze.config.os_version
-  next unless test_platform.downcase == actual_platform && actual_version >= test_version.to_f
+Then(/^on (iOS|macOS|watchOS), (.+)/) do |platform, step_text|
+  step(step_text) if platform_matches?(platform)
+end
 
-  step(step_text)
+Then(/^on (iOS|macOS|watchOS) ([0-9.]+) and later, (.+)/) do |platform, version, step_text|
+  step(step_text) if platform_matches?(platform) && version >= Maze.config.os_version.to_s
```

Comment:
I don't think you want to be doing string comparison for the versions:
```
irb(main):007:0> a = "10"
=> "10"
irb(main):008:0> b = "9"
=> "9"
irb(main):009:0> a >= b
=> false
irb(main):010:0> b >= a
=> true
```

Reply:
Fixed 👍